analytics/management/commands/populate_realms_db.py

```diff
@@ -0,0 +1,201 @@
+                },
+                {
+                    "unique_id": f"{plan_name}-annual-plus",
+                    "sponsorship_pending": False,
+                    "billing_schedule": CustomerPlan.ANNUAL,
+                    "tier": CustomerPlan.PLUS,
+                },
+                {
+                    "unique_id": f"{plan_name}-annual-enterprise",
+                    "sponsorship_pending": False,
+                    "billing_schedule": CustomerPlan.ANNUAL,
+                    "tier": CustomerPlan.ENTERPRISE,
+                },
+                {
+                    "unique_id": f"{plan_name}-monthly-standard",
+                    "sponsorship_pending": False,
+                    "billing_schedule": CustomerPlan.MONTHLY,
+                    "tier": CustomerPlan.STANDARD,
+                },
+                {
+                    "unique_id": f"{plan_name}-monthly-plus",
+                    "sponsorship_pending": False,
+                    "billing_schedule": CustomerPlan.MONTHLY,
+                    "tier": CustomerPlan.PLUS,
+                },
+                {
+                    "unique_id": f"{plan_name}-monthly-enterprise",
+                    "sponsorship_pending": False,
+                    "billing_schedule": CustomerPlan.ANNUAL,
+                    "tier": CustomerPlan.ENTERPRISE,
+                },
+            ]
+
+            # create a realm for each customer profile
+            for i, customer_profile in enumerate(customer_profiles):
+                unique_id = customer_profile["unique_id"]
+                # Delete existing realm with this name
+                try:
+                    get_realm(unique_id).delete()
+                except Realm.DoesNotExist:
+                    pass
+
+                realm = do_create_realm(
+                    string_id=unique_id,
+                    name=unique_id,
+                    description=unique_id,
+                    plan_type=plan_type,
+                )
+
+                # Create a user with billing access
+                full_name = f"{plan_name}-admin"
+                email = f"{full_name}@zulip.com"
+                user = create_user(
+                    email,
+                    full_name,
+                    realm,
+                    full_name,
+                    role=UserProfile.ROLE_REALM_OWNER,
+                )
+
+                administrators_user_group = UserGroup.objects.get(
+                    name=UserGroup.ADMINISTRATORS_GROUP_NAME, realm=realm, is_system_group=True
+                )
+                stream = Stream.objects.create(
+                    name="all",
+                    realm=realm,
+                    can_remove_subscribers_group=administrators_user_group,
+                )
+                recipient = Recipient.objects.create(type_id=stream.id, type=Recipient.STREAM)
+                stream.recipient = recipient
+                stream.save(update_fields=["recipient"])
```

Comment:
please use `create_stream_if_needed`, aka the `do_create_stream` analog, here; we should never ever write duplicate code manually creating `Recipient` objects, even for testing logic; every place we've done that ends up being a pain sometime down the line.

Reply:
Sorry, done.